src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -0,0 +1,1345 @@
+        private const string DirectorySeparatorCharAsString = "/";
+#endif // !PLATFORM_UNIX
+
+        // Platform specific alternate directory separator character.  
+        // This is backslash ('\') on Unix, and slash ('/') on Windows 
+        // and MacOS.
+        // 
+#if !PLATFORM_UNIX        
+        public static readonly char AltDirectorySeparatorChar = '/';
+#else
+        public static readonly char AltDirectorySeparatorChar = '\\';
+#endif // !PLATFORM_UNIX
+
+        // Platform specific volume separator character.  This is colon (':')
+        // on Windows and MacOS, and slash ('/') on Unix.  This is mostly
+        // useful for parsing paths like "c:\windows" or "MacVolume:System Folder".  
+        // 
+#if !PLATFORM_UNIX
+        public static readonly char VolumeSeparatorChar = ':';
+#else
+        public static readonly char VolumeSeparatorChar = '/';
+#endif // !PLATFORM_UNIX
+
+        // Trim trailing white spaces, tabs etc but don't be aggressive in removing everything that has UnicodeCategory of trailing space.
+        // String.WhitespaceChars will trim aggressively than what the underlying FS does (for ex, NTFS, FAT).    
+        internal static readonly char[] TrimEndChars = { (char)0x9, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20, (char)0x85, (char)0xA0 };
+
+#if !PLATFORM_UNIX
+        private static readonly char[] RealInvalidPathChars = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31 };
+
+        // This is used by HasIllegalCharacters
+        private static readonly char[] InvalidPathCharsWithAdditionalChecks = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31, '*', '?' };
+
+        private static readonly char[] InvalidFileNameChars = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31, ':', '*', '?', '\\', '/' };
+#else
+        private static readonly char[] RealInvalidPathChars = { '\0' };
+
+        // This is used by HasIllegalCharacters
+        private static readonly char[] InvalidPathCharsWithAdditionalChecks = { '\0', '*', '?' };
+
+        private static readonly char[] InvalidFileNameChars = { '\0', '*', '?', '\\', '/' };
```

Comment:
Yes, this is .NET Native-only for now, although I think we will want to switch over to this for CoreCLR eventually. We can probably switch over to it immediately because for this particular type there's not really any risk removing it from mscorlib (since it's just a static helper class).